src/debuginfo.cpp

```diff
@@ -1039,7 +1040,7 @@ extern "C" void jl_register_fptrs(uint64_t sysimage_base, void **fptrs, jl_metho
 template<typename T>
 static inline void ignoreError(T &err)
 {
-#if JL_LLVM_VERSION >= 30900 && !defined(NDEBUG)
+#if JL_LLVM_VERSION >= 30900 && !defined(JL_NDEBUG)
```

Comment:
IIUC, this macro is explicitly testing for whether the body is going to resolve either to a no-op or an llvm assertion in the destructor if this hasn't been called.